lib/utils/report.mjs

```diff
@@ -146,3 +134,83 @@ function printfLike(format, ...args) {
 		return result.replace(/%[ds]/, String(arg));
 	}, format);
 }
+
+/** @typedef {import('stylelint').DisabledRangeObject} DisabledRangeObject */
+/** @typedef {import('stylelint').StylelintPostcssResult} StylelintPostcssResult */
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('stylelint').Problem} Problem */
+
+/**
+ * @param {string} ruleName
+ * @param {number} startLine
+ * @param {DisabledRangeObject} [disabledRanges]
+ */
+function isDisabled(ruleName, startLine, disabledRanges) {
+	if (!disabledRanges) return false;
+
+	const ranges = disabledRanges[ruleName] ?? disabledRanges[RULE_NAME_ALL] ?? [];
+
+	for (const range of ranges) {
+		if (
+			// If the problem is within a disabledRange,
+			// and that disabledRange's rules include this one
+			range.start <= startLine &&
+			(range.end === undefined || range.end >= startLine) &&
+			(!range.rules || range.rules.includes(ruleName))
+		) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/** @param {Problem} problem */
+function isFixApplied(problem) {
```

Comment:
[suggest] Not required, but I recommend explicitly having a return type. It's clear and prevents unintentional bugs (e.g. returning undefined).

```suggestion
/**
 * @param {Problem} problem
 * @returns {boolean}
 */
function isFixApplied(problem) {
```

Reply:
I think that's not necessary if the function follows the convention `hasFoo` or `isFoo`.
Returning `undefined` would be falsy so it wouldn't be a big deal.